### ISSUE TITLE
Wiki Link to Attachment fails except with target="_blank"

### ISSUE THREAD

**Issue by gabeos** [taigaio/taiga-front#549]
It seems difficult to link to an attachment on a wiki page from the text of the wiki itself. 

After uploading the attachment and getting the direct link to the file, I tried

```
[Awesome File](https://url.org/media/attachments/a/7/d/5/525b2a1f08df16d76e02c1fa74a7939966d3c3edb3dbed97edda4026a7c7/attachment.pdf)
```

This fails when the link is clicked on, redirecting to taiga 404. However, when the link is opened "In a new tab.." (FFX) it worked just fine.

Ended up specifying the link as:
```html
<a href="https://url.org/media/attachments/a/7/d/5/525b2a1f08df16d76e02c1fa74a7939966d3c3edb3dbed97edda4026a7c7/attachment.pdf" target="_blank">Name Here</a>
```
Seems theres perhaps a small issue in how angular is processing the wiki links here..?

**Comment by superalex** [comment]
Hello @gabeos 

I've tested this here -> https://tree.taiga.io/project/alejandroalonso-test-1/wiki/home and it's working properly if I click on the link (chrome or firefox) allways it's opened in a new tab and working properly.

Could you check if it please?

**Comment by gabeos** [comment]
@superalex hmm, yes your version is working. 

I recently updated to the latest version, any pointers on where to poke around?

**Comment by superalex** [comment]
Hmmmmm, 

No idea, the link you showed seems like well generated, could it be a problem from your browser trying to do something special with pdf's?, have you got the same problem with an image or a text file?

**Comment by Xaviju** [comment]
@gabeos There were problems with **Safari** and _blank and tabs links. Are you using Safari? That should be fixed in this version.

**Comment by gabeos** [comment]
@superalex Nope, seems to be the case with .jpg files as well
@Xaviju Nope this is on up-to-date firefox, fedora 22.

**Comment by superalex** [comment]
@gabeos could you check what exact version are you using? (git describe --tags)

**Comment by gabeos** [comment]
@superalex 
taiga-back: 1.8.0-2-gd6b3a8b
taiga-front: 1.5.0-20-g17a6ab8

**Comment by superalex** [comment]
It seems your taiga-front is a bit outdated...it should be 1.8 too :)

**Comment by gabeos** [comment]
ah, yeah, my heart dropped a bit when i saw that.. :/
then i rebuilt from the htdvisser docker image (again) (--no-cache) and restarted the image.. same version.
i think perhaps its a misunderstanding on how docker-compose works? It used to automatically recreate with the new images, but the recent breaking changes (https://github.com/docker/compose/releases) in compose >= v1.3.0 have possibly changed that behavior...

**Comment by superalex** [comment]
Hmmmmm,

I don't really know how docker is updating it...you could allways use the manual process -> http://taigaio.github.io/taiga-doc/dist/setup-production.html#faq-maintain-update

**Comment by gabeos** [comment]
yeah, the problem with manually updating in a docker image is that it would be overwritten on image restart. the best practice is, i believe, to create the images such that they can be run/restarted wherever and get the same environment. 
Looking into the label issue now..

**Comment by superalex** [comment]
Ok!, so I'm closing this one ;)

Regards!,

**Comment by gabeos** [comment]
Actually, err, maybe you can help me understand this one, relating to the manual upgrade procedure..

```bash
[admin@library ~]$ git clone https://github.com/taigaio/taiga-front-dist.git tfd
Cloning into 'tfd'...
remote: Counting objects: 1084, done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 1084 (delta 6), reused 0 (delta 0), pack-reused 1060
Receiving objects: 100% (1084/1084), 8.41 MiB | 1.46 MiB/s, done.
Resolving deltas: 100% (641/641), done.
Checking connectivity... done.
[admin@library ~]$ cd tfd/
[admin@library tfd]$ git describe --tags
1.5.0-39-gf2e54c5
[admin@library tfd]$ git checkout stable
Branch stable set up to track remote branch stable from origin.
Switched to a new branch 'stable'
[admin@library tfd]$ git describe --tags
1.5.0-20-g17a6ab8
[admin@library tfd]$ git pull
Already up-to-date.
[admin@library tfd]$ git describe --tags
1.5.0-20-g17a6ab8
```

**Comment by superalex** [comment]
Hmmmmm, taiga-front-dist is the compiled version of taiga-front but it seems we are not tagging correctly .

I've updated the correct tags, if you pull again it should be fixed.

**Comment by gabeos** [comment]
okay, thanks for the fix. I updated the front end (in docker) via same procedure, and now have
```bash
[admin@library taiga]$ docker exec -ti taiga_taigafront_1 bash
root@community:/# cd /usr/local/taiga/taiga-front-dist/
root@community:/usr/local/taiga/taiga-front-dist# git describe --tags
1.8.0
```
but have same problem with links :/

**Comment by superalex** [comment]
I don't know what can be happening...I also rebuilt taiga-front-dist and it's exactly the same we are serving on tree.taiga.io, if you are receving a 404 it seems your nginx isn't finding that file for some reason. Could you check your javascript console to see what request it's exactly doing?

**Comment by gabeos** [comment]
yeah, no problem.. I tried that before opening the issue, but oddly didn't see any traffic.. will check again. Maybe nginx config?  will post it to gist in a sec.

**Comment by gabeos** [comment]
https://gist.github.com/gabeos/738c5bcad08689c68e99

**Comment by gabeos** [comment]
aHA! I think I may have caught it! Was only looking at Network Monitor before, should have been in console :dizzy_face: 

Seems that there is a font missing.. It isn't in the dist/fonts directory.

![taiga-font404](https://cloud.githubusercontent.com/assets/751112/8325470/dde2e5b8-1a0e-11e5-9815-7a87baef98bb.png)

**Comment by superalex** [comment]
But that font missing should be related to this one...could you check the console while you click on the link?

**Comment by gabeos** [comment]
wait.. never mind seems that error appeared on the reload page prior to clicking link.. I only get 304's back when I click the link and it takes me to the error page..

**Comment by Xaviju** [comment]
I will add the missing font issue to our bugs list but I don't think this is related to this issue.

**Comment by gabeos** [comment]
wait.. never mind seems that error appeared on the reload page prior to clicking link.. I only get 304's back when I click the link and it takes me to the error page..

**Comment by gabeos** [comment]
Okay, so I restarted my browser, went back to the page, which logged console messages until the missing font, but page loaded just fine. clicked on the link: no messages logged. 404 page shown.
on reloading the wiki page, got to same point, then clicking on link gave me two 304's for the 404 page images.

**Comment by gabeos** [comment]
honestly at this point, a) it's not a huge problem for us, and b) if it would help, I think its reasonable for me to give you temporary access to our site to debug it. We're not storing military secrets, just a non-profit radio station.. I can set up a temp project for you to work on.

**Comment by superalex** [comment]
If you can enable that temporary access I think it's the best option :+1: 

You could write to support@taiga.io with the data

**Comment by gabeos** [comment]
cool. while i do that, i tested in the chrome dev console as well and got an error message that didn't show up in ffx. not sure if it is relevant, but here it is:
![taiga-jquery-err](https://cloud.githubusercontent.com/assets/751112/8325871/a026a586-1a11-11e5-8c83-07b4f2eba53b.png)

**Comment by gabeos** [comment]
Invite sent. If I make you admin on the project, will you get access to the rest of the site?

**Comment by superalex** [comment]
Hello @gabeos,

I think I found the problem, when having links in your wiki pages we have to distinguish between those which are internal (angular handle's them) and the external (we add a target="blank") to process them as "real" links.

We make this distinction based on the initial part of the url, in tree.taiga.io we serve the media files in a different domain (media.taiga.io) so the target="_blank" is properly inserted. In your example you the media url and the site url is the same so we can't add the target="_blank" and angular shows that 404 error page. 

I added the same link for the image without https, using http so Taiga can properly add the target="_blank" attribute.

In the wiki page we are sharing now there are three links, your original one, the same without http and another one with an external image. The last two are working properly.

Regards!,

**Comment by gabeos** [comment]
Ahhh.. interesting. Thanks! Good to know. I hope it was useful tracking that down in some way. Small bug, but I appreciate you spending the time on it (and just making taiga in general. you have no idea how many project management tools I've set up that didn't work out for us in some way. we don't have a budget to pay for commercial tools, so I have set up nearly all of the open source ones I could find. Initially was trying to reduce all our different tools into one thing, but in the end, I like taiga's focused approach. Got us on slack with a non-profit account, set up a separate calendar, and simple time tracking for volunteers, and we're set. Huge huge benefit for us :) Also was able to easily script auto-add to projects for new slack team members. yeah.. anyway. :heart:  )

tune in sometime, we have super diverse programming! -- hollowearthradio.org/listen
